### DISCUSSION TITLE
Storing a queries' where statement for on-demand execution

### DISCUSSION THREAD

**Discussion by psteinroe** [supabase/supabase#8090] [Questions]
Hey!

I am currently looking into solutions on how to store a query into the database and execute it on demand. As an use case, you can imagine customers and customer segments, which are defined by the user as filters, e.g. all customers with an age > 20.

My best guess is to serialise the query into json and store it in the database. The sql statement can then be generated from the json structure on demand.

I could not find much prev. art on such a problem, although I imagine it to be a quite common one. So far, the main inspiration is pg_graphql's query generation. Any inspiration is highly appreciated - also happy to share the results.

Thanks!

**Comment by GaryAustin1** [comment]
Assuming you mean save a query to use in the client code like with supabase-js.
There are probably numerous ways but I do something like this.
Note I used an object to store the filters so I could easily change the object based on user input on a bunch of filter dialogs. An array probably works also.
I don't store it to the database, but it could be turned to json and stored.

```
filterObj = {}
function setFilter(filter) {
        filterObj[filter.col] = filter
    }
setFilter({col:"search_title",op:"like",val:"Cat"})
setFilter({col:"year",op:"eq",val:2019})
```
```
let query = supabase.from(table)
        .select('*')
        .eq('uid', userUid)
Object.keys(filterObj).forEach((key) => {
            let filter = filterObj[key]
            console.log('filter',filter.op,filter)
            switch (filter.op) {
                case "eq":
                    query = query.eq(filter.col, filter.val.toLowerCase())
                    break
                case "contains":
                    query = query.contains(filter.col, [filter.val.toLowerCase()])
                    break
                case "like":
                    query = query.ilike(filter.col, `%${filter.val.toLowerCase()}%`)
                    break
                case "between":
                    query = query.gte(filter.col, filter.val[0])
                    query = query.lte(filter.col, filter.val[1])
            }
        })
let { data: notes, error,count } = await query
```

**Reply by psteinroe** [reply]
Thanks for your reply! Storing it for client side (as in supabase-js) usage is definitely an option,  but I am looking to find a way to execute it from within the database too, e.g. as part of a cron job.
